### PR TITLE
[ZEPPELIN-679] only print the error message for SQL exceptions

### PR DESCRIPTION
**Pull request by zhongneu**
### What is this PR for?

Most of the SQL exceptions are syntax errors. Printing the full stack trace is not necessary. We should only print the error message to avoid distracting the users

### What type of PR is it?
Improvement

### What is the Jira issue?
[ZEPPELIN-679](https://issues.apache.org/jira/browse/ZEPPELIN-679)

### Screenshots (if appropriate)
screen shot 2016-02-26 at 11 44 30 pm

### Questions:
* Does the licenses files need update?
NO

* Is there breaking changes for older versions?
NO

* Does this needs documentation?
NO

### CODE REVIEW

File: spark/src/main/java/org/apache/zeppelin/spark/SparkSqlInterpreter.java [hunk 1]
```diff
@@ -138,6 +138,10 @@ public InterpreterResult interpret(String st, InterpreterContext context) {
       rdd = sqlMethod.invoke(sqlc, st);
     } catch (NoSuchMethodException | SecurityException | IllegalAccessException
         | IllegalArgumentException | InvocationTargetException e) {
+      if (e instanceof InvocationTargetException) {
+        return new InterpreterResult(Code.ERROR,
+                ((InvocationTargetException) e).getTargetException().getMessage());
+      }
```
Comment (felixcheung) [inline spark/src/main/java/org/apache/zeppelin/spark/SparkSqlInterpreter.java]: would it be clearer if it is
```
 } catch (NoSuchMethodException | SecurityException | IllegalAccessException
          | IllegalArgumentException e) {
         throw new InterpreterException(e);
 } catch (InvocationTargetException e) {
         return new InterpreterResult(Code.ERROR,
                 ((InvocationTargetException) e).getTargetException().getMessage());
 }
```

**Comment by zhongneu** [comment]
@felixcheung definitely. updated based on your comment

**Comment by felixcheung** [comment]
thx - let's add logger to capture the full stack before returning with InterpreterResult?
```
logger.error("Invocation target exception", e);
```
?

**Comment by zhongneu** [comment]
@felixcheung great idea. updated & tested. full stack trace will be logged in the interpreter log

**Comment by felixcheung** [comment]
LGTM

**Comment by felixcheung** [comment]
btw, it looks like this might be a legitimate test failure, could you take a look?
https://s3.amazonaws.com/archive.travis-ci.org/jobs/112358128/log.txt
```
Failed tests: 
  SparkSqlInterpreterTest.test:110 Exception not catched
```

**Comment by zhongneu** [comment]
fixed the test

**Comment by echarles** [comment]
I often use the stacktraces to debug what's going wrong. With this, I expect I will miss information. Can we make this configurable (default being "not verbose")?

**Comment by zhongneu** [comment]
@echarles I thought about made this configurable, but I thought the stacktraces might not be helpful for SQL queries, because typically it is difficult for the users to dig into the internal Spark implementation. Could you give me examples of using stacktraces to debug SQL queries?

**Comment by echarles** [comment]
@zhongneu I will rather give you a message a user could face running the SQL: `java.lang.IllegalArgumentException: requirement failed` What will the guy responsible to troubleshoot this ticket will do without the stacktrace?

**Comment by zhongneu** [comment]
@echarles Thanks, Eric! It is good to know we have such cases. We should definitely leave it as configurable if Spark doesn't provide informative error message. I am not sure, but probably in older versions of Spark, there may be more such cases.

Just curious, what kind of queries can result in such exception?

**Comment by nemccarthy** [comment]
this seems a little misguided. hiding errors is great until you need to stack trace. 

id vote to have it configurable where the default is always full stack traces are visible.

**Comment by joelvanveluwen** [comment]
+1 for making it configurable. I find the stack trace invaluable!

**Comment by echarles** [comment]
So we all agree to make it configurable on the interpreter settings leve

`zeppelin.spark.sql.stacktrace = false` by default ?

What about when you catch and retrhow the expection to append a message like "Configure zeppelin.spark.sql.stacktrace = true to see the stacktrace, or use sqlc.sql(query)"

We would have best of both worlds.

@ zhongneu don't have the query at hand, anyway, the stacktrace helped me to identify an bug unrelated to sql but related to storage access.

**Comment by zhongneu** [comment]
I agree to make it configurable, but show the shortened message by default

@echarles I think that's a good suggestion.

**Comment by zhongneu** [comment]
Added a configuration to show full stacktrace

**Comment by Leemoonsoo** [comment]
LGTM

**Comment by felixcheung** [comment]
LGTM. @echarles what do you think?

**Comment by echarles** [comment]
Tested with zeppelin.spark.sql.stacktrace being false and true, and it works fine.

Good to merge.

PS: With HEAD source, I have issues settings any property via the UI (When I restart, the changed values are not taken into account) - Had to set them changing the interpreter.json. Is this only me on my env?

**Comment by AhyoungRyu** [comment]
@echarles I faced the same issue. I think it's a bug..

**Comment by echarles** [comment]
@AhyoungRyu I have opened ZEPPELIN-725 for the interpreter update issue.

@felixcheung @Leemoonsoo This PR #754 is good to merge.

**Comment by AhyoungRyu** [comment]
@echarles Thanks for creating the issue!

**Comment by Leemoonsoo** [comment]
Thanks i'm merging it into master.

**Pull request closed without merging**